include/libdnf/rpm/package_query.hpp

```diff
@@ -587,8 +587,9 @@ class PackageQuery : public PackageSet {
     // TODO(dmach): consider removing the installed packages during the filtering
     void filter_priority();
 
-    /// Filter packages, which are installed but not available in any enabled repository
-    void filter_extras();
+    /// Filter packages, which are installed but not available in any enabled repository.
+    /// @param with_versions If false (default) extras calculation is based only on `name.arch`. That means package is not in extras if any version of the package exists in the repository. If true, filter_extras is more strict and returns each package which exact NEVRA is not present in any repository.
+    void filter_extras(const bool with_versions=false);
```

Comment:
Nice, technically it is not a version but EVR.

Just a suggestion. What about to name the variable as `exact_evr` or `exact_EVR`?

Reply:
Naming, the hardest part of the coding...
I agree, your suggested name for parameter is clearer.